jalib/jassert.cpp

```diff
@@ -46,6 +43,9 @@ int jassert_quiet = 0;
 static int theLogFileFd = -1;
 static int errConsoleFd = -1;
 
+static jalib::string& tmpDir() {static jalib::string s;return s;};
+static jalib::string& uniquePidStr() {static jalib::string s;return s;};
```

Comment:
@karya0:  Here is one possible comment to add in front of the definition of `tmpDir()`.  Feel free to modify it, if you have something better.  Whatever comment we use will then be re-used in occurrences of this paradigm in other files.
`// This function is a "wrapper" around a static local variable.  We will use the function
 // on the left-hand side of an assignment statement.  We cannot promote the static local
 // variable to a static global variable and use the variable on the left-hand side, because
 // in C++, the default constructor would then be called even before main().  That default
 // constructor might call a DMTCP wrapper function, even before DMTCP is fully initialized.`